### DISCUSSION TITLE
Is there a way to limit access to a secret to a specific action?

### DISCUSSION THREAD

**Discussion by stefan-wenig** [community/community#27068] [Actions]
<p>The idea is to ensure that a certain secret is only used in a specific way, which a speicifc custom action can guarantee.</p>
<p>However, I cannot find a way to pass a secret to an action unless the secret is visible to the entire repo. Am I missing something? Is there some advanced secret-fu that would achieve this? If not, any plans?</p>
<p>A comparable feature would be system credentials in Jenkins: they are not accessible by mere build scripts, only by plugins explicitly installed by the administrator.</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="stefan-wenig" data-post="1" data-topic="163471">
<div class="title">
<div class="quote-controls"></div>
 stefan-wenig:</div>
<blockquote>
<p>However, I cannot find a way to pass a secret to an action unless the secret is visible to the entire repo. Am I missing something?</p>
</blockquote>
</aside>
<p>You aren’t, actions can’t access secrets unless the workflow explicitly provides them. As far as I’m aware that’s a design decision for security, so 3rd party actions can’t sneakily access secrets.</p>
